src/plugins/intel_cpu/src/nodes/reorder.cpp

```diff
@@ -103,6 +104,9 @@ void Reorder::initSupportedPrimitiveDescriptors() {
 
     // must be to initialize here since shapes are unknown at the time of Reorder node creation
     isDynamic = !(config.inConfs[0].getMemDesc()->isDefined() && config.outConfs[0].getMemDesc()->isDefined());
+    if (!shapeInference) {
```

Comment:
Not sure I get the point, Here you create shapeInference object w/o any conditions on dynamic shapes. What prevents us from doing that inside the ctor always w/o check on dynamic shapes?

Reply:
Actually nothing except common sense. More over implementations have access to the object and may replace it or call directly The latter is the cause of weakening access rights, because in some nodes we need to properly process padding.
So, the point is, at the level of the second constructor we have no idea on whether the node is dynamic or not, and whether should be instantiate the shape infer class or not. But on the implementation level we at least have this knowledge and hope that the implementation will use it properly.

Reply:
BTW, in this particular line of code, additional condition that the node is dynamic has been added.